includes/amp-helper-functions.php

```diff
@@ -741,9 +742,16 @@ function amp_add_amphtml_link() {
 	}
 
 	$amp_url = amp_add_paired_endpoint( amp_get_current_url() );
+
 	if ( $amp_url ) {
-		$amp_url = remove_query_arg( QueryVar::NOAMP, $amp_url );
-		printf( '<link rel="amphtml" href="%s">', esc_url( $amp_url ) );
+		$amp_url          = remove_query_arg( QueryVar::NOAMP, $amp_url );
+		$sandboxing_level = amp_get_sandboxing_level();
+
+		if ( 1 === $sandboxing_level || 2 === $sandboxing_level ) {
+			printf( '<link rel="alternate" type="text/html" media="only screen and (max-width: 640px)" href="%s">', esc_url( $amp_url ) );
```

Comment:
I wonder if this tag should be extracted into a helper function and reused by both this function and `MobileRedirection`. This deduplication would make it easier to later perhaps introduce a filter for the media query if desired.

Reply:
> This deduplication would make it easier to later perhaps introduce a filter for the media query if desired.

It makes sense. Will add a new helper function to add the alternate link.